lib/model/settings.dart

```diff
@@ -473,3 +474,54 @@ class GlobalSettingsStore extends ChangeNotifier {
     notifyListeners();
   }
 }
+
+/// Whether to show an animated image in its still or animated version.
+///
+/// Use [resolve] to evaluate this for the given [BuildContext],
+/// which reads device-setting data for [animateConditionally].
+///
+/// Callers should try to check whether an image is animated,
+/// i.e. whether it has separate still and animated versions.
+/// This can't be done perfectly (in 2025-10)
+/// because of animated emoji that were uploaded before Zulip Server 5
+/// and don't have `still_url` filled in:
+///   https://github.com/zulip/zulip/issues/36339 .
```

Comment:
I'm not sure what this is saying a caller needs to do.

Is the "can't be done perfectly" just a reference to the fact that ImageEmojiDisplay.resolvedStillUrl is nullable? In that case, the second sentence seems like it belongs more on that field. Or perhaps better yet, on RealmEmojiItem.stillUrl which drives it.